src/PlutoEditor.ts

```diff
@@ -0,0 +1,347 @@
+import { accessSync, readFileSync } from 'fs';
+import { tmpdir } from 'os';
+import { join } from 'path';
+import { v4 as uuid } from 'uuid';
+import * as vscode from 'vscode';
+import { PlutoBackend } from './backend';
+import { getWebviewOptions, LOADING_HTML } from './extension';
+import { create_proxy } from './ws-proxy';
+
+export class PlutoEditor implements vscode.CustomTextEditorProvider {
+
+	public static register(context: vscode.ExtensionContext): vscode.Disposable {
+		const provider = new PlutoEditor(context);
+		const providerRegistration = vscode.window.registerCustomEditorProvider(PlutoEditor.viewType, provider);
+		return providerRegistration;
+	}
+
+	private static readonly viewType = 'plutoView';
+	private readonly pluto_asset_dir = join(tmpdir(), uuid())
+	private readonly webviews = new WebviewCollection();
+	private readonly uri_to_notebook_id_map = new Map <vscode.Uri, string>();
```

Comment:
We have `uri_to_notebook_id_map` but `webviews` also contains this information, since we store the notebook id and uri for each webview. We could remove the `uri_to_notebook_id_map` and rewrite that functionality by searching `webviews`

Reply:
Right now it is unclear if they are added, updated and removed at the same time, or if there are differences between the two.